### PR TITLE
Update cereal to 1.3.2 and minor refresh of documentation and package.

### LINKED ISSUE #19: Upgrade to cereal 1.3.2
Hi @wush978,

Would it be possible to upgrade the version of cereal to v1.3.2? 

Due to new CRAN policies, prior to submitting the updated version of cereal, the following fix would be needed on `base64.hpp`

```sh
sed -i 's|#pragma|// #pragma|' cereal-1.3.2/include/cereal/external/base64.hpp 
```

If you don't have time or interest in managing the package, I'm more than happy to be a maintainer as it's being used in `mlpack`'s _R_ bindings.

Best,

JJB

### PR DESCRIPTION
**Pull request by stephematician**
General update to package:

-   Bumps cereal to 1.3.2.
-   Fixes #19 using a `configure` and `configure.win` script to comment out dis-allowed `#pragma`.
-   Adds instructions to README and vignette for using cpp11.
-   Adds a test for cpp11 usage.
-   Updates the tests to testthat v3: removes deprecated `context` calls.
    -    Removes install_github test as this doesn't test part of the package.
-   Adds dots to `update_version` so that, for example, the library location can be specified by the user.
-   Added common changelog.

__NOTE: CRAN will complain about missing link if release tag 1.3.2 doesn't exist on github__.

__NOTE: Did not test/confirm CI workflow__.

Results from `devtools::check()`: 

```
── R CMD check results ────────────────────────── Rcereal 1.3.2 ────
Duration: 23.4s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔
```
Report from `revdepcheck::revdep_check()`:

```md
## revdepcheck results

We checked 1 reverse dependencies, comparing R CMD check results across CRAN and dev versions of this package.

 * We saw 0 new problems
 * We failed to check 0 packages
```

### CODE REVIEW

**Comment by stephematician** [comment]
Hold - I have pushed the wrong branch.

**Comment by stephematician** [comment]
Correct branch now.

**Comment by wush978** [comment]
Thanks.

I am checking the updated package. It might take me about 2 days I guess.

**Comment by stephematician** [comment]
The markdown files might need a bit of re-formatting, but I thought I'd leave that up to you.

**Comment by wush978** [comment]
Hi @stephematician,

Thank you so much for your work on this PR! Before I merge it, could you kindly add yourself as a contributor to the list of authors?

**Comment by stephematician** [comment]
Thanks, happy to help!

Before you submit to CRAN, make sure to add a tag `1.3.2` on github - otherwise CRAN will complain about a missing URL in `NEWS.md`.

File: tests/testthat/cpp/test_struct_cpp11.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include <sstream>
+
+#include <cpp11/raws.hpp>
+
+#include <cereal/archives/binary.hpp>
+
+[[cpp11::linking_to("Rcereal")]]
```
Comment (stephematician) [inline tests/testthat/cpp/test_struct_cpp11.cpp:7]: I noticed that this fails in workflow: https://github.com/wush978/Rcereal/actions/runs/10894121138/job/30230306285

Perhaps the attribute should be moved to in front of a function (see https://godbolt.org/z/Pdfa8hqqc).

**Review by stephematician — COMMENTED** [review]
Need to fix the example and test using cpp11 for stricter compilers like clang.

**Pull request merged**